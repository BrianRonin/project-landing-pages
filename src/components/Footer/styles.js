import styled, { css } from 'styled-components'
import { Main as Text } from '../Text/styles'
import { Main as SectionContainer } from '../SectionContainer/styles'

export const Main = styled.div`
  ${({ theme }) => css`
    text-align: center;
    font-size: ${theme.fonts.sizes.small};
    border-top: 0.1rem solid ${theme.colors.mediumGray};
    background-color: white;
    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.fonts.sizes.small};
    }

    & ${Text} {
      font-size: ${theme.fonts.sizes.small};
    }

    & ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }
  `}
`
