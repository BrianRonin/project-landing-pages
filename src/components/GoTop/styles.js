import styled, { css } from 'styled-components'

const Light = (theme) => css`
  color: ${theme.colors.primaryColor};
  background: ${theme.colors.white};
`

export const Main = styled.a`
  ${({ theme, background }) => css`
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    right: 2rem;
    bottom: 2rem;
    z-index: 6;
    opacity: 0.8;
    border-radius: 50%;
    ${background && Light(theme)};
  `}
`
