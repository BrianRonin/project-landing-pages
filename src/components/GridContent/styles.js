import styled, { css } from 'styled-components'
export const Main = styled.div`
  ${({ theme }) => css`
    text-align: center;
    align-items: center;
    margin: 0 auto;
    width: 100%;
  `}
`
export const Container_Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.xhuge} 0;
  `}
`
