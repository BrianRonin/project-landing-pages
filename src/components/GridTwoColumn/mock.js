export const mock_GridTwoColumn = {
  children: 'children',
  title: 'titulo',
  text: 'Lorem, ipsum dolor sit amet consectetur adipisicing elit. Doloribus repellat temporibus dicta facilis fugiat corporis adipisci molestias beatae voluptate iure voluptates iste qui possimus accusamus, laudantium aspernatur animi aliquam. Minima.',
  srcImg: 'assets/images/analytics_re_wxt8.svg',
}
