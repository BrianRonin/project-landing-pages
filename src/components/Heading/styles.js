import styled, { css } from 'styled-components'

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.sizes.large};
  `,
  big: (theme) => css`
    font-size: ${theme.sizes.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.sizes.xhuge};
    ${mediaFont(theme)}
  `,
}

const mediaFont = (theme) => css`
  @media ${theme.media.medium} {
    font-size: ${theme.sizes.xlarge};
  }
`

export const Title = styled.h1`
  ${({ theme, light, size, uppercase }) => css`
    color: ${light ? theme.colors.white : theme.colors.primaryColor};
    text-transform: ${uppercase ? 'uppercase' : 'none'};
    ${titleSize[size](theme)};
  `}
`
