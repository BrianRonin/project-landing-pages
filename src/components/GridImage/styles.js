import styled, { css } from 'styled-components'
import { Title } from '../Heading/styles'
import { Main as Text } from '../Text/styles'
export const Main = styled.div`
  ${({ theme }) => css`
    > ${Text} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`
export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(30rem, 1fr));
    gap: ${theme.spacings.large};
    width: 100%;

    @media ${theme.media.medium} {
      grid-template-columns: 1fr;
    } ;
  `}
`

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
    max-width: 100%;
  `}
`

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    transition: all 300ms ease-in-out;
    &:hover {
      transform: scale(1.2) rotate(10deg);
    }
  `}
`
