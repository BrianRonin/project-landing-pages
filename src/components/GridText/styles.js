import styled, { css } from 'styled-components'
import { Title } from '../Heading/styles'
import { Main as Text } from '../Text/styles'
export const Main = styled.div`
  ${({ theme }) => css``}
`
export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    ${Text} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${Title} {
      position: relative;
      left: 5rem;
    }

    ${Title}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 7rem;
      top: -3rem;
      left: -5rem;
      transform: rotate(10deg);
    }
  `}
`
