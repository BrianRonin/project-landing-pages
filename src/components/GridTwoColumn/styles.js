import styled, { css } from 'styled-components'
import { Title } from '../Heading/styles'
export const Main = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.sizes.large};

    @media ${theme.media.medium} {
      grid-template-columns: 1fr;
    }

    ${Title} {
      margin-bottom: ${theme.sizes.xlarge};
    }
  `}
`
export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.medium} {
      margin-bottom: ${theme.sizes.large};
    }
  `}
`

export const ImageContainer = styled.div`
  ${({ theme }) => css``}
`

export const Img = styled.img`
  ${({ theme }) => css``}
`
