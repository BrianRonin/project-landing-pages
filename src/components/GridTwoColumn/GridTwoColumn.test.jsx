import { renderTheme } from '../../styles/render-theme'
import { GridTwoColumn } from './index'
import { mock_GridTwoColumn } from './mock'

describe('<GridTwoColumn />', () => {
  it('should render two column grid and to match snapshot', () => {
    const { container } = renderTheme(<GridTwoColumn {...mock_GridTwoColumn} />)
    expect(container).toMatchSnapshot()
  })
})
