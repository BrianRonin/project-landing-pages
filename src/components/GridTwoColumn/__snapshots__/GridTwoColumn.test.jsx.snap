// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoColumn /> should render two column grid and to match snapshot 1`] = `
.c5 {
  color: #0A1128;
  text-transform: uppercase;
  font-size: 6.4rem;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr 2fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3.2rem;
}

.c2 .c4 {
  margin-bottom: 4.0rem;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c6 {
  font-size: 2.4rem;
}

@media (max-width:768px) {
  .c5 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c2 {
    grid-template-columns: 1fr;
  }
}

@media (max-width:768px) {
  .c3 {
    margin-bottom: 3.2rem;
  }
}

<div>
  <div
    class="c0"
  >
    <section
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h2
            class="c4 c5"
          >
            titulo
          </h2>
          <div
            class="c6"
          >
            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Doloribus repellat temporibus dicta facilis fugiat corporis adipisci molestias beatae voluptate iure voluptates iste qui possimus accusamus, laudantium aspernatur animi aliquam. Minima.
          </div>
        </div>
        <div
          class=""
        >
          <img
            alt="titulo"
            class=""
            src="assets/images/analytics_re_wxt8.svg"
          />
        </div>
      </div>
    </section>
  </div>
</div>
`;
